hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadTableState.java

```diff
@@ -83,11 +96,20 @@ public int getOperationPos() {
   }
 
   public int[] getRequiredPositions() {
-    final List<String> fieldNames = rowType.getFieldNames();
-    return requiredRowType.getFieldNames().stream()
-        .map(fieldNames::indexOf)
-        .mapToInt(i -> i)
-        .toArray();
+    List<String> fieldNames = rowType.getFieldNames();
+    List<String> expColumns = new ArrayList<>();
+    int[] pos = requiredRowType.getFieldNames().stream()
+            .mapToInt(fn -> {
+              if (!fieldNames.contains(fn.toLowerCase())) {
+                expColumns.add(fn);
+              }
+              return fieldNames.indexOf(fn.toLowerCase());
+            })
+            .toArray();
+    if (!expColumns.isEmpty()) {
+      throw new HoodieException("Column(s) " + String.join(", ", expColumns) + " does not exists in the hudi table " + this.tableName + ".");
+    }
```

Comment:
Column(s) [`$col_a`, `$col_b`, `$col_c` ...] does not exist in the table `$tableName`.

Reply:
Done.

Reply:
Should we move the check into `HoodieTableFactory#createDynamicTableSource` ?